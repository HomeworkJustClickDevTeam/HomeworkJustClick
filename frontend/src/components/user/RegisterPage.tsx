import React, {ChangeEvent, useState} from "react"
import {postAuthRegisterPostgresService} from "../../services/postgresDatabase"
import {useNavigate} from "react-router-dom"
import KolkoLewe from './left_circle.svg';
import Smile from './Smile.svg';
import {UserRegisterInterface} from "../../types/UserRegisterInterface";


const RegisterPage = () => {
  const [user, setUser] = useState<UserRegisterInterface>({
    firstname: "",
    lastname: "",
    password: "",
    secondPassword: "",
    email: ""
  })
  const navigate = useNavigate()

  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    if (!!user.password && user.password === user.secondPassword) {
      try {
        await postAuthRegisterPostgresService(user)
        navigate("/")
      } catch (e) {
        console.log("Error not send")
      }
    }
  }
  const handleChangeUser = (event: ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.target
    setUser((prevState) => ({
      ...prevState,
      [name]: value,
    }))
  }

  return (
      <div className='flex w-screen flex-col font-lato font-normal text-sm select-none'>
        <img className="fixed left-[4%]  bottom-[6%] scale-50 xl:scale-100 -z-50" src = {KolkoLewe} alt="Kółko po lewej stronie"></img>
        <div className='flex justify-center items-center text-center flex-col'>
          <h1 className='mt-16 mb-16 text-4xl xl:text-[64px] ml-[7%]'>Dołącz do nas!
            <img className="relative right-0  scale-50 translate-x-[25%] xl:transform-none -z-50 inline-block pl-10" src= {Smile} alt="Kółko smile"></img>
          </h1>
          {/*<img className="absolute right-0 top-0 mr-[24%] mt-[8%]" src= {Smile} alt="Kółko smile"></img>*/}
          <form onSubmit={handleSubmit} className='flex flex-col text-center items-center'>
            <input
                type="text"
                name="firstname"
                value={user.firstname}
                placeholder="Imię"
                onChange={handleChangeUser}
                className='border-b-2 border-b-light_gray mb-12 text-center placeholder:text-light_gray placeholder:text-[12px] w-60 xl:text-[20px] xl:placeholder:text-[20px] xl:w-[320px]'
            />
            <input
                type="text"
                name="lastname"
                value={user.lastname}
                placeholder="Nazwisko"
                onChange={handleChangeUser}
                className='border-b-2 border-b-light_gray mb-12 text-center placeholder:text-light_gray placeholder:text-[12px] w-60 xl:text-[20px] xl:placeholder:text-[20px] xl:w-[320px]'
            />
            <input
                type="email"
                name="email"
                value={user.email}
                placeholder="Adres e-mail"
                onChange={handleChangeUser}
                className='border-b-2 border-b-light_gray mb-12 text-center placeholder:text-light_gray placeholder:text-[12px] w-60 xl:text-[20px] xl:placeholder:text-[20px] xl:w-[320px]'
            />
            <input
                type="password"
                name="password"
                value={user.password}
                placeholder="Hasło"
                onChange={handleChangeUser}
                className='border-b-2 border-b-light_gray mb-12 text-center placeholder:text-light_gray placeholder:text-[12px] w-60 xl:text-[20px] xl:placeholder:text-[20px] xl:w-[320px]'
            />
            <input
                type="password"
                name="secondPassword"
                value={user.secondPassword}
                placeholder="Powtórz hasło jeszcze raz"
                onChange={handleChangeUser}
                className='border-b-2 border-b-light_gray mb-12 text-center placeholder:text-light_gray placeholder:text-[12px] w-60 xl:text-[20px] xl:placeholder:text-[20px] xl:w-[320px]'
            />
            <button type="submit" className=' bg-main_blue w-40 h-8 rounded-[5px] xl:rounded-[10px] xl:w-[288px] xl:h-[56px] text-white xl:text-[28px] hover:bg-hover_blue hover:shadow-md active:shadow-none'>Zajerestruj się</button>
          </form>
        </div>
      </div>
  )
}
export default RegisterPage
