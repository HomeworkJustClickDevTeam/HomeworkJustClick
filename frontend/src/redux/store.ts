import {combineReducers, configureStore, PreloadedState} from "@reduxjs/toolkit"
import userStateReducer from "./userStateSlice"
import roleReducer from "./roleSlice"
import groupReducer from "./groupSlice"
import isLoadingReducer from "./isLoadingSlice"
import homePageInReducer from "./homePageInSlice"



const rootReducer = combineReducers({
  userState: userStateReducer,
  role: roleReducer,
  group: groupReducer,
  isLoading: isLoadingReducer,
  homePageIn: homePageInReducer
})
export const setupStore = (preloadedState?: PreloadedState<RootState>) =>{
  return configureStore({
    reducer: rootReducer,
    preloadedState
  })
}


export type RootState = ReturnType<typeof rootReducer>
export type AppStore = ReturnType<typeof setupStore>
export type AppDispatch = AppStore['dispatch']