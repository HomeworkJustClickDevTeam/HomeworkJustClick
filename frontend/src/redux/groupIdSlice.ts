import {createSlice, PayloadAction} from "@reduxjs/toolkit";
import {RootState} from "./store";

type GroupIdSliceType = number | null
const initialState = null as GroupIdSliceType

export const groupIdSlice = createSlice({
    name: 'groupId',
    initialState,
    reducers: {
        setGroupId: (state, action: PayloadAction<GroupIdSliceType>) => action.payload
    }
})

export const {setGroupId} = groupIdSlice.actions
export const selectGroupId = (state: RootState) => state.groupId
export default groupIdSlice.reducer