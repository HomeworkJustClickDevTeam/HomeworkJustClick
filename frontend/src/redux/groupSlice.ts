import { createSlice, PayloadAction } from "@reduxjs/toolkit"
import { GroupInterface } from "../types/GroupInterface"
import { RootState } from "./store"

type GroupSliceType = GroupInterface | null
const initialState = null as GroupSliceType
export const groupSlice = createSlice({
  name:'group',
  initialState,
  reducers:{
    setGroup:(state, action:PayloadAction<GroupSliceType>) => action.payload
  }
})

export const {setGroup} = groupSlice.actions
export const selectGroup = (state:RootState) => state.group
export default groupSlice.reducer