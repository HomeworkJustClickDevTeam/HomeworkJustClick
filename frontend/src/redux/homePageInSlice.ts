import { createSlice, PayloadAction } from "@reduxjs/toolkit"
import { RootState } from "./store"

const initialState: boolean = true
export const homePageInSlice = createSlice({
  name: 'homePageIn',
  initialState,
  reducers:{
    setHomePageIn:(state, action:PayloadAction<boolean>) => action.payload
  }
})

export const {setHomePageIn} = homePageInSlice.actions
export const selectHomePageIn = (state:RootState) => state.homePageIn
export default homePageInSlice.reducer