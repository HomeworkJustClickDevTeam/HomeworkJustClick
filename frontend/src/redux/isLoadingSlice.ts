import { createDraftSafeSelector, createSlice } from "@reduxjs/toolkit"
import { AppDispatch, RootState } from "./store"

const initialState: boolean = false

export const isLoadingSlice = createSlice({
  name: 'isLoading',
  initialState,
  reducers:{
    setIsLoading: (_, action) => action.payload
  }
})

export const {setIsLoading} = isLoadingSlice.actions

export const setIsLoadingInReducer = (newState:boolean) => {
  return (dispatch:AppDispatch) => {
    dispatch(setIsLoading(newState))
  }
}
export const selectIsLoading = (state:RootState):boolean => state.isLoading //Use when you want to obtain the state of isLoading
export default isLoadingSlice.reducer