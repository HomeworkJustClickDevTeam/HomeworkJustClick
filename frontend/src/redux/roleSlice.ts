import { createSlice, PayloadAction } from "@reduxjs/toolkit"
import { RootState } from "./store"


type RoleSliceType = string|null
const initialState = null as RoleSliceType

export const roleSlice = createSlice({
  name: 'role',
  initialState,
  reducers:{
    setRole:(state, action:PayloadAction<RoleSliceType>) => action.payload
  }
})

export const {setRole} = roleSlice.actions
export const selectRole = (state:RootState) => state.role
export default roleSlice.reducer