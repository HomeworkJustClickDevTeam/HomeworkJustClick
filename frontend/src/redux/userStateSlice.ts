import { createSlice, PayloadAction } from "@reduxjs/toolkit"
import { UserInterface } from "../types/UserInterface"
import { RootState } from "./store"

type UserStateSliceType = UserInterface | null
export const userStateSlice = createSlice({
  name: 'userState',
  initialState: null as UserStateSliceType,
  reducers: {
    logOut: () => null,
    setUser: (state, action: PayloadAction<UserStateSliceType>) => action.payload
  }
})

export const {logOut, setUser} = userStateSlice.actions
export const selectUserState = (state: RootState) => state.userState
export default userStateSlice.reducer