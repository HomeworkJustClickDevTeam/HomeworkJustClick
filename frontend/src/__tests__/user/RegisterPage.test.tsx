import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { setupStore } from "../../redux/store";
import { BrowserRouter } from "react-router-dom";
import HomeGuestPage from "../../components/home/HomeGuestPage";
import RegisterPage from "../../components/user/RegisterPage";



it("checks if register page renders correctly", () => {
  const renderedHomeGuestPage = renderer
    .create(
      <Provider store={setupStore()}>
        <BrowserRouter>
          <RegisterPage></RegisterPage>
        </BrowserRouter>
      </Provider>
    )
  expect(renderedHomeGuestPage).toMatchSnapshot()
});