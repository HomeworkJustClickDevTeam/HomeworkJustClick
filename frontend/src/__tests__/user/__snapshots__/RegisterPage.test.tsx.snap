// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if register page renders correctly 1`] = `
<div
  className="flex w-screen flex-col font-lato font-normal text-sm select-none"
>
  <img
    alt="Kółko po lewej stronie"
    className="fixed left-[4%]  bottom-[6%] scale-50 xl:scale-100 -z-50"
    src={
      {
        "process": [Function],
      }
    }
  />
  <div
    className="flex justify-center items-center text-center flex-col"
  >
    <h1
      className="mt-16 mb-16 text-4xl xl:text-[64px] ml-[7%]"
    >
      Dołącz do nas!
      <img
        alt="Kółko smile"
        className="relative right-0  scale-50 translate-x-[25%] xl:transform-none -z-50 inline-block pl-10"
        src={
          {
            "process": [Function],
          }
        }
      />
    </h1>
    <form
      className="flex flex-col text-center items-center"
      onSubmit={[Function]}
    >
      <input
        className="border-b-2 border-b-light_gray mb-12 text-center placeholder:text-light_gray placeholder:text-[12px] w-60 xl:text-[20px] xl:placeholder:text-[20px] xl:w-[320px]"
        name="firstname"
        onChange={[Function]}
        placeholder="Imię"
        type="text"
        value=""
      />
      <input
        className="border-b-2 border-b-light_gray mb-12 text-center placeholder:text-light_gray placeholder:text-[12px] w-60 xl:text-[20px] xl:placeholder:text-[20px] xl:w-[320px]"
        name="lastname"
        onChange={[Function]}
        placeholder="Nazwisko"
        type="text"
        value=""
      />
      <input
        className="border-b-2 border-b-light_gray mb-12 text-center placeholder:text-light_gray placeholder:text-[12px] w-60 xl:text-[20px] xl:placeholder:text-[20px] xl:w-[320px]"
        name="email"
        onChange={[Function]}
        placeholder="Adres e-mail"
        type="email"
        value=""
      />
      <input
        className="border-b-2 border-b-light_gray mb-12 text-center placeholder:text-light_gray placeholder:text-[12px] w-60 xl:text-[20px] xl:placeholder:text-[20px] xl:w-[320px]"
        name="password"
        onChange={[Function]}
        placeholder="Hasło"
        type="password"
        value=""
      />
      <input
        className="border-b-2 border-b-light_gray mb-12 text-center placeholder:text-light_gray placeholder:text-[12px] w-60 xl:text-[20px] xl:placeholder:text-[20px] xl:w-[320px]"
        name="secondPassword"
        onChange={[Function]}
        placeholder="Powtórz hasło jeszcze raz"
        type="password"
        value=""
      />
      <button
        className=" bg-main_blue w-40 h-8 rounded-[5px] xl:rounded-[10px] xl:w-[288px] xl:h-[56px] text-white xl:text-[28px] hover:bg-hover_blue hover:shadow-md active:shadow-none"
        type="submit"
      >
        Zajerestruj się
      </button>
    </form>
  </div>
</div>
`;
